File: frontend/models/Reminder.ts
/**
 * Reminder entity: represents one notification reminder attached to an appointment.
 */
export class Reminder {
  constructor(
    public reminderId: number,
    public remindAt: Date,
    public method: string,
  ) {}

  getReminderInfo(): string {
    return `${this.method} at ${this.remindAt.toLocaleString()}`;
  }
}

File: CalendarApi/Models/Reminder.cs
namespace CalendarApi.Models;

public enum ReminderMethod
{
    POPUP,
    EMAIL,
    SMS
}

/// <summary>
/// Reminder entity: represents one notification reminder attached to an appointment.
/// </summary>
public class Reminder
{
    public int ReminderId { get; set; }
    public DateTime RemindAt { get; set; }
    public ReminderMethod Method { get; set; }

    public Reminder(int reminderId, DateTime remindAt, ReminderMethod method)
    {
        ReminderId = reminderId;
        RemindAt = remindAt;
        Method = method;
    }

    public string GetReminderInfo()
    {
        return $"{Method} at {RemindAt.ToLocalTime()}";
    }
}
